### PR TITLE
Add prompt for /home partition

### PR DESCRIPTION
**Pull request by HTV04**
Adds the following prompt to the archinstall setup: "Would you like to create a separate partition for /home? (Y/n): "

This allows the user to choose whether to add a separate partition for /home (current default behavior) or include it within the root partition, even if BTRFS subvolumes aren't being used or there is enough space on the disk for a separate /home partition.

### CODE REVIEW

**Comment by dylanmtaylor** [comment]
We already have fully custom partitioning support, I'm not sure if this is that much more user-friendly.

**Comment by HTV04** [comment]
> We already have fully custom partitioning support, I'm not sure if this is that much more user-friendly.

It's a common choice to make a separate /home partition because it allows for easy persistence between Arch reinstalls, however, the same goes for those who would prefer to keep everything in one partition. Having a /home partition means that the root partition must be limited, and the script by default limits it to 20 GB, giving the rest to the /home partition.

While archinstall does support custom partition support (via scripting, iirc), having this option available from the installer would still be very useful and convenient, in my opinion.

**Review by Torxed — CHANGES_REQUESTED** [review]

File: archinstall/lib/disk/user_guides.py [hunk 1]
```diff
@@ -45,10 +46,14 @@ def suggest_single_disk_layout(block_device, default_filesystem=None, advanced_o
 		}
 	})
 
+	if not using_subvolumes or block_device.size >= MIN_SIZE_TO_ALLOW_HOME_PART:
```
[the remark this answers is not in the document]
Comment (HTV04) [reply archinstall/lib/disk/user_guides.py]: Fixed in 31a00ca

**Comment by Torxed** [comment]
Thank you for those latest changes.
In recent light of all the trouble the `/home` partition has created, I think it makes sense to merge this in so that users can more safely choose whether or not they want a separate `/home` partition and judge it for themselves. Manually editing the JSON every time they want to remove/add it is a bit cumbersome and this saves some time.

**Pull request merged**